sample/src/main/java/io/github/boguszpawlowski/composecalendar/sample/CustomComponentsSample.kt

```diff
@@ -24,23 +27,37 @@ import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import io.github.boguszpawlowski.composecalendar.StaticCalendar
 import io.github.boguszpawlowski.composecalendar.day.NonSelectableDayState
-import io.github.boguszpawlowski.composecalendar.header.MonthState
+import io.github.boguszpawlowski.composecalendar.header.DefaultWeekDaysNames
+import io.github.boguszpawlowski.composecalendar.rememberCalendarState
+import io.github.boguszpawlowski.composecalendar.states.CurrentState
+import io.github.boguszpawlowski.composecalendar.states.EventState
 import java.time.DayOfWeek
 import java.time.DayOfWeek.SUNDAY
 import java.time.format.TextStyle.NARROW
 import java.util.Locale
 
 @Composable
 fun CustomComponentsSample() {
-  StaticCalendar(
-    modifier = Modifier.animateContentSize(),
-    showAdjacentMonths = false,
-    firstDayOfWeek = SUNDAY,
-    monthContainer = { MonthContainer(it) },
-    dayContent = { DayContent(dayState = it) },
-    weekHeader = { WeekHeader(daysOfWeek = it) },
-    monthHeader = { MonthHeader(monthState = it) },
-  )
+  val scrollState = rememberScrollState()
+  Column(modifier = Modifier
+    .fillMaxWidth()
+    .verticalScroll(scrollState)
+  ) {
+    val calendarState = rememberCalendarState(
+      eventState = EventState( dayEventList )
+    )
+    ModeControls(modeState = calendarState.modeState)
+    StaticCalendar(
+      modifier = Modifier.animateContentSize(),
+      showAdjacentMonths = false,
+      firstDayOfWeek = SUNDAY,
+      monthContainer = { MonthContainer(it) },
+      dayContent = { DayContent(dayState = it) },
+      weekDaysNames = { DefaultWeekDaysNames(listDays = it) },
+      monthHeader = { MonthHeader(currentState = it) },
+      calendarState = calendarState
+    )
+  }
```

Comment:
This cuould be in a separate `WeekCalendar` sample

Reply:
I see that you are doing it in the next fork as you see fit. Tomorrow or the day after tomorrow I will have time to look at your varant in detail.